treeherder/seta/analyze_failures.py

```diff
@@ -0,0 +1,141 @@
+            high_value_jobs = get_high_value_jobs(revisions_fixed_by_commit_plus_tagged_jobs,
+                                                  self.analysis_percentage,
+                                                  self.ignore_failures)
+
+            if not self.dry_run:
+                self.clear_expiration_field_for_expired_jobs()
+                self.increase_jobs_priority(high_value_jobs)
+
+    def get_raw_data(self):
+        """Reach the data/seta endpoint for all failures that have been marked as "fixed by commit"
+
+        The endpoint returns a dictionary with a key per revision or bug ID (the bug ID is used for
+        intermittent failures and the revision is used for real failures). The failures for *real
+        failures* will contain all jobs that have been starred as "fixed by commit".
+
+        Notice that the raw data does not tell you on which repository a root failure was fixed.
+
+        For instance, in the raw data you might see a reference to 9fa614d8310d which is a back out
+        and it is reference by 12 starred jobs:
+            https://treeherder.mozilla.org/#/jobs?repo=autoland&filter-searchStr=android%20debug%20cpp&tochange=9fa614d8310db9aabe85cc3c3cff6281fe1edb0c
+        The raw data will show those 12 jobs.
+
+        We return the obtained data or an empty structure if we failed to fetch it.
+
+        [1]
+        {
+          "failures": {
+            "44d29bac3654": [
+              ["android-4-0-armv7-api15", "opt", "android-lint", 2804 ],
+              ["android-4-0-armv7-api15", "opt", "android-api-15-gradle-dependencies", 2801],
+        """
+        # XXX: Once this uses data from Treeherder we will be able to query the table directly
+        url = self.seta_host + "/data/seta/?startDate=%s&endDate=%s" % (
+            self.start_date, self.end_date)
+        LOG.info('Grabbing information from {}'.format(url))
+        try:
+            response = retry(requests.get,
+                             args=(url, ),
+                             kwargs={
+                                 'headers': {
+                                     'Accept': 'application/json',
+                                     'User-Agent': 'treeherder',
+                                 },
+                                 'verify': True
+                             })
+            data = json.loads(response.content)
+        except Exception as error:
+            # XXX: Should we not simply abort the script?
+            # we will return an empty 'failures' list if got exception here
+            LOG.warning("The request for %s failed due to %s" % (url, error))
+            LOG.warning("We will work with an empty list of failures")
+            data = {'failures': []}
+
+        LOG.info("start-date: {}, end-date: {}, failures: {}".format(
+             self.start_date, self.end_date, len(data['failures'])))
+        return data['failures']
+
+    def clear_expiration_field_for_expired_jobs(self):
```

Comment:
You might want to define a manager class for JobPriority and include this method as a member for it. Then you could do something like:

JobPriority.objects.clear_expiration_field_for_expired_jobs()

Which makes it a bit more clear that we're doing a table-level operation. You can read up on manager methods here:

https://docs.djangoproject.com/en/1.10/topics/db/managers/#custom-managers

There's lots of examples of this in `treeherder/model/models.py`. Just search for "manager" in this file.